forge-game/src/main/java/forge/game/replacement/ReplacementHandler.java

```diff
@@ -118,12 +118,10 @@ public List<ReplacementEffect> getReplacementList(final ReplacementType event, f
             public boolean visit(Card crd) {
                 Card c = preList.get(crd);
                 Zone cardZone = game.getZoneOf(c);
-                Zone lkiZone = game.getChangeZoneLKIInfo(c).getLastKnownZone();
 
                 // only when not prelist
-                boolean noLKIstate = c != crd || event != ReplacementType.Moved;
+                boolean noLKIstate = c != crd || event != ReplacementType.Moved || !ZoneType.Battlefield.equals(runParams.get(AbilityKey.Origin));;
```

Comment:
turns out it's not good enough :/
the logic needs to be like before:
- if the RE host was in play, use RE host LKI
but now it's instead:
- if card LTB, use RE host LKI

this breaks the Exile part of _Bleed Dry_

Reply:
Urgh, then it probably would have failed in my old branch too

can you try to fix it on this branch?

Reply:
I think I healed it now